src/runtime/buffer.h

```diff
@@ -92,6 +92,25 @@ static inline void extend_total(buffer b, int offset)
     }
 }
 
+static inline bytes buffer_set_capacity(buffer b, bytes len)
+{
+    assert(!buffer_is_wrapped(b));
+    if (len < b->end - b->start)
+        len = b->end - b->start;
+    if (len != b->length) {
+        void *new = allocate(b->h, len);
```

Comment:
This block is nearly identical to one in buffer_extend...maybe this can be unified / refactored? Maybe buffer_extend and extend_total can use buffer_set_capacity internally. (That interface is confusing to begin with...extend_total should probably be renamed.)

Reply:
OK, I refactored buffer_extend() to use buffer_set_capacity() internally. I didn't change extend_total() because the most "logical" way for extend_total() to call buffer_set_capacity() would be as in `buffer_set_capacity(b, offset)`, but with that change the vector unit tests take forever to complete, because the buffer of a vector gets resized at each vector element insertion; and there might be places in the kernel as well that take advantage of the resize logic that is in buffer_extend(). So I think it's better for extend_total() to keep calling buffer_extend().